miner/worker.go

```diff
@@ -559,36 +626,26 @@ func (w *worker) resultLoop() {
 			if w.chain.HasBlock(block.Hash(), block.NumberU64()) {
 				continue
 			}
-			// Update the block hash in all logs since it is now available and not when the
-			// receipt/log of individual transactions were created.
-			for _, r := range result.receipts {
-				for _, l := range r.Logs {
-					l.BlockHash = block.Hash()
-				}
-			}
-			for _, log := range result.state.Logs() {
-				log.BlockHash = block.Hash()
-			}
 			// Commit block and state to database.
+			// Note same state changes could be committed many times because two different valid blocks
+			// sharing same state.
+			// Except for the first commit, the remaining commits will not write more state data to the database.
```

Comment:
I don't think you need this comment here. How the chain handles state writes should be up to the state. The miner shouldn't really care. Especially since the chain actually does in memory GC, so most of the data will never even hit the disk.